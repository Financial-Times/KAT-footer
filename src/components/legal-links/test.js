import React from 'react';
import { mount, shallow } from 'enzyme';
import LegalLinks from './index.js';

describe('LegalLinks', () => {
  const testData = {
    links: [
      {
        href: "test",
        label: "test"
      },
      {
        href: "test",
        label: "test"
      }
    ]
  };

  const testNull = {
    links: 0
  };

  describe('component', () => {
    const wrapper = mount(<LegalLinks linksData={testData.links} />);

    const content =
    <div>
      <ul className="o-footer__legal-links">
        <li key={0}>
          <a href={testData.links[0].href}>
            {testData.links[0].label}
          </a>
        </li>
        <li key={1}>
          <a href={testData.links[1].href}>
            {testData.links[1].label}
          </a>
        </li>
      </ul>
    </div>;

    test('matches snapshot', () => {
      const wrapper = shallow(<LegalLinks linksData={testData.links} />);
      expect(wrapper.debug()).toMatchSnapshot();
    });

    test('has the right content', () => {
      expect(wrapper.contains(content)).toEqual(true);
    });
  });

  describe('return null if there is no links', () => {
    const wrapper = mount(<LegalLinks linksData={testNull.links} />);

    test('has the right content', () => {
      expect(wrapper.component.state.context).toEqual(null);
    });
  });
});
