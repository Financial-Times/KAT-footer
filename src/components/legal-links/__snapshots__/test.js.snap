// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegalLinks component matches snapshot 1`] = `
'<div>
<ul className=\\'o-footer__legal-links\\'>
<li>
<a href=\\'test\\'>
test
</a>
</li>
<li>
<a href=\\'test\\'>
test
</a>
</li>
</ul>
</div>'
`;
